# Copyright 2017 Robot Garden, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

#
# Settings for cone_finder
#

#Min throttle value during cone seek - actually just the min value
#that Victoria can drive at. For road - 30, small grass - 40 and 
#tall grass - 50 or more
#min_throttle_on_grass: 0.4
min_throttle_on_grass: 0.55
min_throttle_on_road: 0.15
#min_throttle will be set as parameter in code from one of these

#Max throttle needs to be higher on grass than on road
#max_throttle_on_grass: 0.8
max_throttle_on_grass: 1.0
max_throttle_on_road: 0.6
#max_throttle will be set as parameter in code from one of these

#These distances (in ft) are used to control the throttle when approaching cones
# Range4    >[3] => use max throttle
# Range3 [3]-[2] => use throttle between max and (max+min)/2
# Range2 [2]-[1] => use throttle between min and (max+min)/2
# Range1 [1]-[0] => use min throttle
# Range0    <[0] => cut the throttle
throttle_distances_on_grass: [0.7, 6, 12, 20]
throttle_distances_on_road: [1, 12, 20, 25]
#throttle_distances will be set as parameter in code from one of these

# Minimum confidence for cone seeker to assume we have a cone
min_confidence: 0.1

# Height values at different distance
# On a 640x480 size image (realsense), cone height is ~40 pixels @25'
# ~100p @10', ~120p @9', ~130p @8', ~150p @7', ~170@6'
# ~210p @5', ~270@4', ~370@3', ~440 @2', 480@18"-10" and cone starts to cut and area may decrease
height_25ft: 40
height_15ft: 65
height_10ft: 100
height_7ft: 150
height_5ft: 210
height_4ft: 270
height_3ft: 370
height_2ft: 440
height_max: 480

# Params for cutting throttle
# On a 640x480 size image, cone area is ~300 sq pixels @25'
# ~2500sp @10', ~8000sp @6', ~12500sp @5', ~20000sp @4'
# ~34000sp @3' and cone starts to cut and area may decrease
min_area: 300

# Even after the throttle is cut, rover may continue to roll
tc_timeout: 2                    # Cut throttle after this timeout

#Settings when searching for cone
search_period: 8
#search_throttle_on_grass: 0.6
search_throttle_on_grass: 0.7
search_throttle_on_road:  0.3
#search_throttle will be set as parameter in code from one of these

